### DISCUSSION TITLE
HELP: setAmbient does not work in simple sky  node

### DISCUSSION THREAD

**Discussion by XianranZ** [pelicanmapping/osgearth#2382] [Support Q&A] [answered]
Hello, everyone.
I am a newer in osgEarth and OSG. And I am using OSG3.4 and osgEarth2.7.0.
I encounter a strange problem when I want to control the brightness of night in simple sky node, which is the setAmbient function does not work. I don't know why but I am sure that I have set the ambient to 1.0. In theory, the night should become as bright as the day. But it does not happen. Here is my code:
```c++
osgEarth::Util::SkyOptions skyOptions;
skyOptions.ambient = 1.0;
m_skyNode = osgEarth::Util::SkyNode::create (skyOptions, m_mapNode.get());
osgEarth::DateTime date(2023, 11, 14, 11);
m_skyNode->setDateTime(date);
m_skyNode->setLighting(osg::StateAttribute::ON);
osgEarth::insertGroup(m_skyNode.get(), m_mapNode.get());
m_skyNode->attach(m_viewer.get(), 0);
```

In order to find the problem, I further looked at the source code of simple sky node, and then a more strange problem appeared, which is that The change of functions cannot affect the color of the light source. I modified some source code in the function named "void SimpleSkyNode::initialize(const SpatialReference* srs)" in SimpleSkyNode.cpp. The original code is:
```c++
_light->setDiffuse(osg::vec4f(1.0f, 1.0f, 1.0f, 1.0f));
_light->setSpecular(osg::vec4f(1.0f, 1.0f, 1.0f, 1.0f));
```
The modified code is:
```c++
_light->setDiffuse(osg::vec4f(1.0f, 0.0f, 0.0f, 1.0f));
_light->setSpecular(osg::vec4f(1.0f, 0.0f, 0.0f, 1.0f));
```
In theory, the color of sunlight should become red. But it does not happen.
It seem like the sunlight is not controlled by the '_light'. 

 Why?

It is a strange and challenge problem, and I look forward to getting your help.

Thanks a lot!

**Comment by gwaldron** [comment, marked as answer]
Hello.
First, I hope you are using a newer version of osgEarth ... 2.7 is rather old :)
Second, there was indeed a defect (in the master branch) in which the diffuse color was not applied in the shader. I have pushed a fix for this. Ambient already works. Keep in mind however that specular light color is not a factor in the PBR lighting model so that will have no effect. 
Hope this helps.

**Reply by XianranZ** [reply]
Thanks for your help very much!

Due to some force majeure factors, I cannot use the newer version.
So I need to make the setAmbient and setDiffuse work in osgEarth2.7.0.
I will download your fixed codes to make the setDiffuse working.

However, You said the ambient already works. I'm sorry I don't understand you.
You mean it works in the newer version, or it works in osgEarth2.7.0, or I need to download the fixed codes you provided and then it will work. Please make this clear.

Thanks again for your help.

**Reply by gwaldron** [reply]
I meant that it already works in 3.5.

**Reply by XianranZ** [reply]
Ok.
Thanks a lot. 
I will use osgEarth3.5 to try it.

**Reply by gwaldron** [reply]
The fix I said I pushed is post-3.5, so you will need the master branch to get that as well, Good luck.
